iijpta/ngx_http_secure_token_iijpta.c

```diff
@@ -0,0 +1,240 @@
+#include <ngx_config.h>
+
+#include "ngx_http_secure_token_iijpta.h"
+#include "../ngx_http_secure_token_filter_module.h"
+#include "../ngx_http_secure_token_utils.h"
+
+#include <openssl/evp.h>
+
+// constants
+#define CRC32_SIZE    4
+#define EXPIRY_SIZE   8
+#define PATH_LIMIT    1024
+
+// typedefs
+typedef struct {
+	ngx_str_t key;
+	ngx_str_t iv;
+        ngx_http_complex_value_t *acl;
+	ngx_secure_token_time_t end;
+} ngx_secure_token_iijpta_token_t;
+
+typedef struct {
+        u_char crc[CRC32_SIZE];
+        u_char expiry[EXPIRY_SIZE];
+} ngx_http_secure_token_iijpta_header_t;
+
+static ngx_conf_num_bounds_t ngx_http_secure_token_iijpta_key_bounds = {
+	ngx_conf_check_str_len_bounds, 16, 16
+};
+
+static ngx_conf_num_bounds_t ngx_http_secure_token_iijpta_iv_bounds = {
+	ngx_conf_check_str_len_bounds, 16, 16
+};
+
+// globals
+static ngx_command_t ngx_http_secure_token_iijpta_cmds[] = {
+	{ ngx_string("key"),
+	NGX_CONF_TAKE1,
+	ngx_http_secure_token_conf_set_hex_str_slot,
+	0,
+	offsetof(ngx_secure_token_iijpta_token_t, key),
+	&ngx_http_secure_token_iijpta_key_bounds },
+
+	{ ngx_string("iv"),
+	NGX_CONF_TAKE1,
+	ngx_http_secure_token_conf_set_hex_str_slot,
+	0,
+	offsetof(ngx_secure_token_iijpta_token_t, iv),
+	&ngx_http_secure_token_iijpta_iv_bounds },
+
+	{ ngx_string("acl"),
+	NGX_CONF_TAKE1,
+        ngx_http_set_complex_value_slot,
+	0,
+	offsetof(ngx_secure_token_iijpta_token_t, acl),
+	NULL },
+
+	{ ngx_string("end"),
+	NGX_CONF_TAKE1,
+        ngx_http_secure_token_conf_set_time_slot,
+	0,
+	offsetof(ngx_secure_token_iijpta_token_t, end),
+	NULL },
+};
+
+static ngx_int_t
+ngx_http_secure_token_get_acl_iijpta(ngx_http_request_t *r, ngx_http_complex_value_t *acl_conf, ngx_str_t* acl)
+{
+	// get the acl
+	if (acl_conf != NULL)
+	{
+		if (ngx_http_complex_value(r, acl_conf, acl) != NGX_OK)
+		{
+			return NGX_ERROR;
+		}
+	}
+	else
+	{
+		// the default is '/*'
+	        acl->data = (u_char *)"/*";
```

Comment:
can use ngx_str_set (btw, currently there's a bug here, since the sizeof counts the null)